packages/timeline-state-resolver/src/devices/vmixAPI.ts

```diff
@@ -1,91 +1,200 @@
+		const dur = args.duration !== undefined ? `&Duration=${args.duration}` : ''
+		const mix = args.mix !== undefined ? `&Mix=${args.mix}` : ''
+		const ext = args.extra !== undefined ? args.extra : ''
+
+		const queryString = `${inp}${val}${dur}${mix}${ext}`.slice(1) // remove the first &
+		const command = `FUNCTION ${func} ${queryString}`
+
+		// this.emit('debug', `Sending command: ${command}`)
+
+		return this._sendCommand(command)
+	}
+
+	private async _sendCommand(cmd: string): Promise<Error | undefined> {
+		return new Promise<Error | undefined>((r) => {
+			this._socket?.write(cmd + '\r\n', (e) => (e ? r(e) : r(undefined)))
 		})
 	}
 
-	private async _connectHTTP(options?: VMixOptions): Promise<boolean> {
-		if (options) {
-			if (!(options.host.startsWith('http://') || options.host.startsWith('https://'))) {
-				options.host = `http://${options.host}`
+	private async _processIncomingData(data: Buffer) {
+		const string = data.toString('utf-8')
+		const newLines = string.split('\r\n')
+
+		this._unprocessedLines.push(...newLines)
+
+		while (this._unprocessedLines.length > 0) {
+			const result = RESPONSE_REGEX.exec(this._unprocessedLines[0])
+			let processedLines = 0
+
+			if (result && result.groups?.['response']) {
+				// create a response object
+				const responseLen = parseInt(result?.groups?.['response'])
+				const response: Response = {
+					command: result.groups?.['command'],
+					response: (Number.isNaN(responseLen) ? result.groups?.['response'] : 'OK') as Response['response'],
+					message: result.groups?.['responseMsg'],
+					body: undefined as undefined | string,
+				}
+				processedLines++
+
+				// parse additional lines if needed
+				if (!Number.isNaN(responseLen)) {
+					let len = responseLen
+					const lines: string[] = []
+
+					while (len > 0) {
+						const l = this._unprocessedLines[lines.length + 1] // offset of 1 because first line is not data
+						if (!l) break // todo - should we wait for streaming data to come in?
+
+						len -= l.length + 2
+						lines.push(l)
+					}
+					response.body = lines.join('\r\n')
+					processedLines += lines.length
+				}
+
+				// now do something with response
+				this.emit('data', response)
+			} else {
+				// well this is not happy, do we do something?
+				processedLines++
+			}
+
+			// remove processed lines
+			this._unprocessedLines.splice(0, processedLines)
+		}
+	}
+
+	private _triggerReconnect() {
+		if (!this._reconnectTimeout) {
+			this._reconnectTimeout = setTimeout(() => {
+				this._reconnectTimeout = undefined
+
+				if (!this._connected) this._setupSocket()
+			}, 5000)
+		}
+	}
+
+	private _setupSocket() {
+		if (this._socket) {
+			this._socket.removeAllListeners()
+			if (!this._socket.destroyed) {
+				this._socket.destroy()
 			}
-			this._options = options
 		}
 
-		return new Promise((resolve) => {
-			this.once('initialized', () => {
-				this.emit('stateChanged', this.state)
-				resolve(true)
-			})
+		this._socket = new Socket()
```

Comment:
I think this should use `this._socket.setNoDelay(true)` (https://nodejs.org/api/net.html#socketsetnodelaynodelay)